File: lib/times.py
import time
from crud_operations import crud_operations

def time_read_function():
    CRUD = crud_operations()
    
    start_time = time.time()
    data = CRUD.read()
    end_time = time.time()
    
    print(f"data_head: {data.head()}")  # Adjusted for YAML format
    print(f"python_read_time: {end_time - start_time:.6f}")

if __name__ == "__main__":
    time_read_function()

File: src/times.rs
use std::time::Instant;
mod crud;

fn main() {
    let start = Instant::now();

    // Run the read operation from your existing CRUD module
    let universities = crud::read().unwrap();
    
    let duration = start.elapsed();
    // Print the timing and a sample of the data as YAML-like output
    println!("data_head: {:?}", universities.get(0)); // Sample head of data
    println!("rust_read_time: {:.6}", duration.as_secs_f64());
}
